modules/rke2/main.tf

```
@@ -0,0 +1,166 @@
+locals {
+  uname = var.unique_suffix ? lower("${var.name}-${random_string.uid.result}") : lower(var.name)
+}
+
+resource "random_string" "uid" {
+  length  = 3
+  special = false
+  lower   = true
+  upper   = false
+  numeric = true
+}
+
+resource "random_password" "token" {
+  length  = 40
+  special = false
+}
+
+data "nutanix_cluster" "cluster" {
+  name = var.nutanix_cluster
+}
+
+data "nutanix_subnet" "subnet" {
+  subnet_name = var.nutanix_subnet
+}
+
+data "nutanix_image" "image" {
+  image_name = var.image_name
+}
+
+resource "nutanix_virtual_machine" "rke2_bootstrap" {
+  name         = "${local.uname}-server-0"
+  cluster_uuid = data.nutanix_cluster.cluster.id
+
+  memory_size_mib      = var.server_memory
+  num_sockets          = var.server_cpu
+  num_vcpus_per_socket = var.server_cpu_cores
+
+  disk_list {
+    data_source_reference = {
+      kind = "image"
+      uuid = data.nutanix_image.image.id
+    }
+    device_properties {
+      disk_address = {
+        device_index = 0
+        adapter_type = "SCSI"
+      }
+      device_type = "DISK"
+    }
+    disk_size_mib = var.server_primary_disk_size
+  }
+  disk_list {
+    disk_size_mib = var.server_secondary_disk_size
+  }
+
+  nic_list {
+    subnet_uuid = data.nutanix_subnet.subnet.id
+    dynamic "ip_endpoint_list" {
+      for_each = length(var.server_ip_list) != 0 ? [1] : []
+      content {
+        ip   = var.server_ip_list[0]
+        type = "ASSIGNED"
+      }
+    }
+  }
+
+  guest_customization_cloud_init_user_data = base64encode(templatefile("${path.module}/cloud-config.tpl.yaml", {
+    hostname        = "${local.uname}-server-0",
+    authorized_keys = var.ssh_authorized_keys,
+    token           = random_password.token.result,
+    bootstrap_ip    = var.server_dns_name,
```

```suggestion
    bootstrap_ip    = "",
```
bootstrap node shouldn't be passed a value for this